builder/llpcBuilderImpl.h

```diff
@@ -84,10 +84,33 @@ class BuilderImplArith : virtual public BuilderImplBase
 public:
     BuilderImplArith(LLVMContext& context) : BuilderImplBase(context) {}
 
+    // Create calculation of 2D texture coordinates that would be used for accessing the selected cube map face for
+    // the given cube map texture coordinates.
+    Value* CreateCubeFaceCoord(Value* pCoord, const Twine& instName = "") override final;
+
+    // Create calculation of the index of the cube map face that would be accessed by a texture lookup function for
+    // the given cube map texture coordinates.
+    Value* CreateCubeFaceIndex(Value* pCoord, const Twine& instName = "") override final;
+
+    // Create quantize operation.
+    Value* CreateQuantize(Value* pInValue, Type* pQuantTy, const Twine& instName = "") override final;
+
+    // Create signed integer modulo operation.
+    Value* CreateSMod(Value* pDividend, Value* pDivisor, const Twine& instName = "") override final;
+
 private:
     LLPC_DISALLOW_DEFAULT_CTOR(BuilderImplArith)
     LLPC_DISALLOW_COPY_AND_ASSIGN(BuilderImplArith)
 
+    // Create "isNaN" operation: return true if the supplied FP (or vector) value is NaN
+    Value* CreateIsNaN(Value* pInValue, const Twine& instName = "");
```

Comment:
I already do this in PR #172. Do you really want me to squash that commit into this one? The end result is the same either way.

Reply:
Yes, you can merge the commit to this one. That commit is not big.
Just CreateIsInfinite -> CreateIsInf, pInValue -> pX

Reply:
Actually I will leave it as a separate commit, because I get rebase conflicts moving it next to this commit so I can squash it. So it saves quite a lot of work leaving it separate.

Reply:
It's OK if it is convenient for you.